### DISCUSSION TITLE
Can i choose the render mode depending what browser do the request?

### DISCUSSION THREAD

**Discussion by JavierDaga** [dotnet/aspnetcore#52968] [General] [answered]
I have an webassembly app runing with .NET 7, but i have a problem with safari, simply not work in that version (don't remember what version was). I have a preview version of that project made in webassembly Server Side that has worked in that version of safari, becouse of that i wondering if i can choose de render mode depending what browser do the reques to recover the compatibility with that version of safari.

**Comment by javiercn** [comment, marked as answer]
Yes, you can pass `@rendermode` as an attribute to the component and provide the value dynamically based on whatever criteria you see fit.

```razor
@{
  var renderMode= RenderMode.InteractiveWebassembly;
  if(HttpContext.Headers....){
      renderMode = RenderMode.InteractiveServer
  }
}
...
  <Routes @rendermode="renderMode" />
```
